shared/src/commonMain/kotlin/org/hyperskill/app/learning_activities/data/repository/LearningActivitiesRepositoryImpl.kt

```diff
@@ -46,4 +49,17 @@ class LearningActivitiesRepositoryImpl(
                 types = types
             )
         )
+
+    override suspend fun getLearningActivitiesWithSections(
+        studyPlanSectionTypes: Set<StudyPlanSectionType>,
+        learningActivityTypes: Set<LearningActivityType>,
+        learningActivityStates: Set<LearningActivityState>
```

Comment:
As I remember, we don't have such filtering in new API, the endpoint always returns only TODO activities

Reply:
We have such filtering in the new API